src/app/shared/services/switchboard-toaster.service.ts

```diff
@@ -0,0 +1,76 @@
+import { Injectable } from '@angular/core';
+import { ToastrService } from 'ngx-toastr';
+import { IndividualConfig } from 'ngx-toastr/toastr/toastr-config';
+import { BehaviorSubject, Observable } from 'rxjs';
+
+enum MessageType {
+    error = 'toast-error',
+    info = 'toast-info',
+    success = 'toast-success',
+    warning = 'toast-warning'
+}
+
+export interface SwitchboardToaster {
+    message: string;
+    type: string;
+    isNew: boolean;
+}
+@Injectable({
+    providedIn: 'root'
+})
+export class SwitchboardToasterService {
+    private massageList = new BehaviorSubject<SwitchboardToaster[]>([]);
+
+    constructor(private toaster: ToastrService) {
+    }
+
+    getMessageList(): Observable<SwitchboardToaster[]> {
+        return this.massageList.asObservable();
+    }
+
+    reset(): void {
+        this.massageList.next([]);
+    }
+
+    readAllItems(): void {
+        this.massageList.next(this.massageList.getValue().map(item => {
```

Comment:
What do you think about not mutating data? So it would be
```suggestion
        this.massageList.next(this.massageList.getValue().map(item => ({...item, isNew: false}))
```